spec/controllers/sys_controller_spec.rb

```diff
@@ -28,208 +28,549 @@
 
 require 'spec_helper'
 
-module OpenProjectRepositoryAuthenticationSpecs
```

Comment:
You should either turn this into a request/request spec or considerably reduce the stack you are testing. You are currently not unit testing as you are testing down to the model.

I'd:
* write request specs for the happy case
* write unit test for the controller where you mock like hell
* write unit tests for the policies

That should reduce the amount of testing you are doing.